File: Beetech.Tms.Core/Models/TextileItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Beetech.Tms.Core.Utils;

namespace Beetech.Tms.Core.Models;

public class TextileItem : BaseEntity
{
    /// <summary>
    /// Unique 24 hex digits used for file RFID EPC.
    /// Calculated from ID using ChaCha20 encryption.
    /// </summary>
    [NotMapped]
    public string Code => Id > 0
        ? ChaCha20Util.EncryptExt(Id, RfidTagConstants.AssetTag)
        : string.Empty;

    public int CategoryId { get; set; }

    public ItemStatus Status { get; set; } = ItemStatus.Available;

    public int WashCount { get; set; } = 0;

    public int? CurrentLocationId { get; set; }
    public int? CurrentDepartmentId { get; set; }

    public DateTime LastScanAt { get; set; } = DateTime.UtcNow;

    // Navigation
    [ForeignKey(nameof(CategoryId))]
    public Category Category { get; set; } = null!;

    [ForeignKey(nameof(CurrentLocationId))]
    public Location? CurrentLocation { get; set; }

    [ForeignKey(nameof(CurrentDepartmentId))]
    public Department? CurrentDepartment { get; set; }
}

File: AndroidClient/app/src/main/java/beetech/tms/android/data/models/TextileItem.java
package beetech.tms.android.data.models;

import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.Ignore;
import com.google.gson.annotations.SerializedName;
import java.io.Serializable;
import beetech.app.core.utils.ChaCha20Util;

@Entity(tableName = "items")
public class TextileItem implements Serializable {
    @PrimaryKey
    @SerializedName("id")
    public int id;
    
    @SerializedName("code")
    public String code;
    
    @SerializedName("status")
    public String status;
    
    @SerializedName("category")
    public String category;
    
    @SerializedName("location")
    public String location;
    
    @SerializedName("department")
    public String department;
    
    @SerializedName("washCount")
    public int washCount;
    
    @SerializedName("epc")
    public String epc;

    @Ignore
    private boolean isSelected;

    public boolean isSelected() {
        return isSelected;
    }

    public void setSelected(boolean selected) {
        isSelected = selected;
    }

    public String getExpectedEpc() {
        // AssetTag prefix is 3001
        return ChaCha20Util.encryptExt(id, "3001");
    }
}
